### PR TITLE
Reset Pagination After Applying Filters

### PR DESCRIPTION
**Pull request by bbluechip**

### CODE REVIEW

**Comment by bbluechip** [comment]
in my last commit, code works on some cases... Page numbers reset if selected category contains more than one page and if you are on that page before you choose that category.

File: packages/ui/src/components/CategoryFilter/CategoryFilter.tsx [hunk 1]
```diff
@@ -49,10 +52,20 @@ export const CategoryFilter: FC<CategoryFilterProps> = ({
   useEffect(() => {
     setIsLoading(true)
   }, [value, setIsLoading])
+  
+  const resetPagination = () => {
```
Comment (7alip) [inline packages/ui/src/components/CategoryFilter/CategoryFilter.tsx:56]: It's not only about changing categories. `selectCategories` prop calls changeParam function.
This should be handled within the useChangeParams hook to ensure consistency across all query params updates.

I mentioned here: https://github.com/freedomcombination/monorepo/pull/1670#discussion_r1757455570

**Review by 7alip — CHANGES_REQUESTED** [review]
Thanks

File: packages/ui/src/hooks/useChangeParam.tsx [hunk 2]
```diff
@@ -16,6 +16,10 @@ export const useChangeParams = () => {
     const newQuery = cleanQuery(query, args)
     cleanArgs(args)
 
+    if (args['categories']) {
+      args['page'] = 1
+    }
+
```
Comment (7alip) [inline packages/ui/src/hooks/useChangeParam.tsx:22]: Not only `categories` filter but every possible filter other than `page` should be considered. For example, should the `page` filter should also be removed when people search `(?q=search)`. There can be other filters too.

**Review by 7alip — CHANGES_REQUESTED** [review]
It's not based on the latest dev. There will be conflicts when you try to merge `dev` into your branch.

I would create a new branch from dev.

**Pull request closed without merging**